File: pset6/vigenere/vigenere.py
import sys
import cs50

# encrypts letter using key letter
def shift(char, num_letter):
    shift = ord(sys.argv[1][num_letter].lower()) - 97
    return chr((ord(char) - 97 + shift) % 26 + 97)

# check if argument is correct
if len(sys.argv) == 2 and sys.argv[1].isalpha():
    
    # ask for text for encrypt
    plain_text = input("plaintext: ")
    
    # number of key's letter
    num_letter = 0
    
    result = ''

    # iterates through text's letters
    for char in plain_text:
        if char.isalpha():
            if char.isupper():
                char = char.lower()
                result += shift(char, num_letter).upper()
            else:
                result += shift(char, num_letter)

            if num_letter < len(sys.argv[1]) - 1:
                num_letter += 1
            else:
                num_letter = 0
        else:
            result += char

    print(f"ciphertext: {result}")
else:
    print("Usage: python vigenere.py key")
    sys.exit(1)

File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int shift(char c);

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        string alphabetic = argv[1];
        int key = 0;
        for (int j = 0; j < strlen(alphabetic); j++)
        {
            char letter = alphabetic[j];
            if (!isalpha(letter))
            {
                printf("Usage: ./vigenere keyword\n");
                return 1;
            }
        }
        string plain_text = get_string("plaintext: ");
        printf("ciphertext: ");
        for (int i = 0, k = 0; i < strlen(plain_text); i++)
        {
            if (k == strlen(alphabetic))
            {
                k = 0;
            }
            key = shift(alphabetic[k]);
            if (plain_text[i] >= 65 && plain_text[i] <= 90)
            {
                printf("%c", (65 + ((plain_text[i] % 65) + key) % 26));
                k++;
            }
            if (plain_text[i] >= 97 && plain_text[i] <= 122)
            {
                printf("%c", (97 + ((plain_text[i] % 97) + key) % 26));
                k++;
            }
            if (plain_text[i] < 65 || (plain_text[i] > 90 && plain_text[i] < 97) || plain_text[i] > 122)
            {
                printf("%c", plain_text[i]);
            }
        }
        printf("\n");
    }
    else
    {
        return 1;
    }
}

int shift(char c)
{
    int shift;
    if islower(c)
    {
        shift = (int) c - 97;
    }
    else
    {
        shift = (int) c - 65;
    }
    return shift;
}